lib/src/commands/Deprecations.ts

```diff
@@ -1,7 +1,65 @@
 import once from 'lodash/once';
+import get from 'lodash/get';
+import each from 'lodash/each';
 import { Platform } from 'react-native';
 
 export class Deprecations {
+  private deprecatedMethods: Array<{ key: string; showWarning: any }> = [
+    {
+      key: 'topBar.searchBarHiddenWhenScrolling',
+      showWarning: once((_key: string, parentOptions: object) => {
+        console.warn(
+          `${_key} is deprecated and will be deprecated in the next major version. For more information see https://github.com/wix/react-native-navigation/issues/6585`,
+          parentOptions
+        );
+      }),
+    },
+    {
+      key: 'topBar.searchBarPlaceholder',
+      showWarning: once((_key: string, parentOptions: object) => {
+        console.warn(
+          `${_key} is deprecated and will be deprecated in the next major version. For more information see https://github.com/wix/react-native-navigation/issues/6585`,
+          parentOptions
+        );
+      }),
+    },
+    {
+      key: 'topBar.searchBarBackgroundColor',
+      showWarning: once((_key: string, parentOptions: object) => {
+        console.warn(
+          `${_key} is deprecated and will be deprecated in the next major version. For more information see https://github.com/wix/react-native-navigation/issues/6585`,
+          parentOptions
+        );
+      }),
+    },
+    {
+      key: 'topBar.searchBarTintColor',
+      showWarning: once((_key: string, parentOptions: object) => {
+        console.warn(
+          `${_key} is deprecated and will be deprecated in the next major version. For more information see https://github.com/wix/react-native-navigation/issues/6585`,
+          parentOptions
+        );
+      }),
+    },
+    {
+      key: 'topBar.hideNavBarOnFocusSearchBar',
+      showWarning: once((_key: string, parentOptions: object) => {
+        console.warn(
+          `${_key} is deprecated and will be deprecated in the next major version. For more information see https://github.com/wix/react-native-navigation/issues/6585`,
+          parentOptions
+        );
+      }),
+    },
+  ];
+
+  public checkForDeprecatedOptions(options: Record<string, any>) {
+    each(this.deprecatedMethods, (method) => {
+      if (get(options, method.key, null)) {
+        method.showWarning(method.key, options);
```

Comment:
Rename to `deprecatedOptions` ?
```suggestion
    each(this.deprecatedOptions, (option) => {
      if (get(options, option.key, null)) {
        option.showWarning(option.key, options);
```